src/toast/_libtoast_tod_pointing.cpp

```diff
@@ -36,16 +33,23 @@ void init_tod_pointing(py::module & m) {
                   throw std::runtime_error(o.str().c_str());
               }
               double * rawpdata = reinterpret_cast <double *> (info_pdata.ptr);
-              double * rawhwpang = reinterpret_cast <double *> (info_hwpang.ptr);
               uint8_t * rawflags = reinterpret_cast <uint8_t *> (info_flags.ptr);
               double * rawweights = reinterpret_cast <double *> (info_weights.ptr);
               int64_t * rawpixels = reinterpret_cast <int64_t *> (info_pixels.ptr);
+              double * rawhwpang = NULL;
+              if (!hwpang.is_none()) {
+                  auto hwpbuf = py::cast <py::buffer> (hwpang);
+                  pybuffer_check_1D <double> (hwpbuf);
+                  py::buffer_info info_hwpang = hwpbuf.request();
+                  rawhwpang = reinterpret_cast <double *> (info_hwpang.ptr);
```

Comment:
Shouldn't you check for the length of `hwpang` just like other vectors are checked?